File: src/main/java/com/ject/studytrip/pomodoro/domain/repository/PomodoroRepository.java
package com.ject.studytrip.pomodoro.domain.repository;

import com.ject.studytrip.pomodoro.domain.model.Pomodoro;
import java.util.Optional;

public interface PomodoroRepository {
    Pomodoro save(Pomodoro pomodoro);

    Optional<Pomodoro> findByDailyGoalId(Long dailyGoalId);
}

File: src/main/kotlin/com/ject/studytrip/pomodoro/domain/repository/PomodoroRepository.kt
package com.ject.studytrip.pomodoro.domain.repository

import com.ject.studytrip.pomodoro.domain.model.Pomodoro
import java.util.Optional

interface PomodoroRepository {
    fun save(pomodoro: Pomodoro): Pomodoro

    fun findByDailyGoalId(dailyGoalId: Long): Optional<Pomodoro>
}
